pkg/k8s/configMap.go

```diff
@@ -0,0 +1,88 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// newConfigMap creates a new configMap structure
+func NewConfigMap(
+	namespace string,
+	name string,
+	labels map[string]string,
+	data map[string]string,
+) *corev1.ConfigMap {
+
+	configMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Data: data,
+	}
+
+	return configMap
+}
+
+func CreateConfigMap (
+	ctx context.Context,
+	cl client.Client,
+	configMap *corev1.ConfigMap,
+	logger logr.Logger,
```

Comment:
let's put the logger next to `ctx` to be consistent

```suggestion
	ctx context.Context,
	logger logr.Logger,
	cl client.Client,
	configMap *corev1.ConfigMap,
```